File: rh-front/src/app/models/demandeRequest.ts
export class DemandeRequest {
    id : number;
    dateDebut : Date;
    dateFin : Date;
    natureAbsence : string;
    justificatif : string;
    statut : string;
    employeId : number;

    constructor() {
        this.id = 0;
        this.dateDebut = new Date();
        this.dateFin = new Date();
        this.natureAbsence = '';
        this.justificatif = '';
        this.statut ='';
        this.employeId = 0;
    }
}

File: absence-service/src/main/java/com/example/absenceservice/model/DemandeRequest.java
package com.example.absenceservice.model;

import com.example.absenceservice.entities.NatureAbsence;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.util.Date;

@Data @AllArgsConstructor @NoArgsConstructor
public class DemandeRequest {
    private Long id;
    private Date dateDebut;
    private Date dateFin;
    private NatureAbsence natureAbsence;
    private byte[] justificatif;
    private String statut;
    private Long employeId;
}
